File: app/src/main/java/com/example/bepresent/CalendarDot.java
package com.example.bepresent;



import android.content.Context;
import android.graphics.Color;

import androidx.core.content.ContextCompat;

import com.prolificinteractive.materialcalendarview.CalendarDay;
import com.prolificinteractive.materialcalendarview.DayViewDecorator;
import com.prolificinteractive.materialcalendarview.DayViewFacade;
import com.prolificinteractive.materialcalendarview.spans.DotSpan;
import java.util.Collection;
import java.util.HashSet;

public class CalendarDot implements DayViewDecorator {

    private final int color;
    private final HashSet<CalendarDay> dates;
    private final Context context;

    public CalendarDot(int color, Collection<CalendarDay> dates, Context context) {
        this.color = color;
        this.dates = new HashSet<>(dates);
        this.context = context;
    }

    @Override
    public boolean shouldDecorate(CalendarDay day) {
       boolean lol = dates.contains(day);

        return lol;
    }

    @Override
    public void decorate(DayViewFacade view) {
        view.addSpan(new DotSpan(10, color));

    }
}

File: app/src/main/java/com/example/bepresent/calendar/CalendarDot.kt
package com.example.bepresent.calendar

import android.content.Context
import com.prolificinteractive.materialcalendarview.CalendarDay
import com.prolificinteractive.materialcalendarview.DayViewDecorator
import com.prolificinteractive.materialcalendarview.DayViewFacade
import com.prolificinteractive.materialcalendarview.spans.DotSpan

class CalendarDot(private val color: Int, dates: Collection<CalendarDay>?, context: Context) : DayViewDecorator {
    private val dates: HashSet<CalendarDay>
    private val context: Context

    init {
        this.dates = HashSet(dates)
        this.context = context
    }

    override fun shouldDecorate(day: CalendarDay): Boolean {
        return dates.contains(day)
    }

    override fun decorate(view: DayViewFacade) {
        view.addSpan(DotSpan(5.0F, color))
    }
}